statsmodels/sandbox/predict_functional.py

```diff
@@ -0,0 +1,438 @@
+
+    This works for models fit without a formula.
+
+    Returns
+    -------
+    exog : data frame
+        A data frame in which the focus variable varies and the other variables
+        are fixed at specified or computed values.
+    """
+
+    model = result.model
+    model_exog = model.exog
+    exog_names = model.exog_names
+
+    if summaries is None:
+        summaries = {}
+    if values is None:
+        values = {}
+
+    exog = np.zeros((num_points, model_exog.shape[1]))
+
+    # Check for variables whose values are not set either through
+    # `values` or `summaries`.
+    colnames = list(values.keys()) + list(summaries.keys()) + [focus_var]
+    unmatched = set(exog_names) - set(colnames)
+    unmatched = list(unmatched)
+    if len(unmatched) > 0:
+        warnings.warn("%s in model but not in `summaries` or `values`."
+                      % ", ".join(["'%s'" % x for x in unmatched]))
+
+    # The values of the 'focus variable' are a sequence of percentiles
+    pctls = np.linspace(0, 100, num_points).tolist()
+    ix = exog_names.index(focus_var)
+    fvals = np.percentile(model_exog[:, ix], pctls)
+    exog[:, ix] = fvals
+
+    # The values of the other variables may be given by summary functions...
+    for ky in summaries.keys():
+        ix = exog_names.index(ky)
+        exog[:, ix] = summaries[ky](model_exog[:, ix])
+
+    # or they may be provided as given values.
+    for ky in values.keys():
+        ix = exog_names.index(ky)
+        exog[:, ix] = values[ky]
+
+    return exog, fvals
+
+
+def _make_exog(result, focus_var, summaries, values, num_points):
+
+    # Branch depending on whether the model was fit with a formula.
+    if hasattr(result.model.data, "frame"):
+        dexog, fexog, fvals = _make_exog_from_formula(result, focus_var,
+                                       summaries, values, num_points)
+    else:
+        exog, fvals = _make_exog_from_arrays(result, focus_var, summaries,
+                                 values, num_points)
+        dexog, fexog = exog, exog
+
+    return dexog, fexog, fvals
+
+
+def _check_args(values, summaries, values2, summaries2):
+
+    if values is None:
+        values = {}
+    if values2 is None:
+        values2 = {}
+    if summaries is None:
+        summaries = {}
+    if summaries2 is None:
+        summaries2 = {}
+
+    for (s,v) in (summaries, values), (summaries2, values2):
+        ky = set(v.keys()) & set(s.keys())
+        ky = list(ky)
+        if len(ky) > 0:
+            raise ValueError("One or more variable names are contained in both `summaries` and `values`:" +
+                             ", ".join(ky))
+
+    return values, summaries, values2, summaries2
+
+
+def predict_functional(result, focus_var, summaries=None, values=None,
+                       summaries2=None, values2=None, cvrg_prob=0.95,
+                       method="pointwise", linear=True, num_points=10,
```

Comment:
method will also need an indication that it is for confidence interval confint_method or ci_method. I don't know if we use it already somewhere. It will be coming in several functions or methods.